Modules/mathmodule.c

```diff
@@ -811,7 +811,15 @@ long_lcm(PyObject *a, PyObject *b)
     if (_PyLong_IsZero((PyLongObject *)a) || _PyLong_IsZero((PyLongObject *)b)) {
         return PyLong_FromLong(0);
     }
-    g = _PyLong_GCD(a, b);
+
+    /* Make sure a <= b to speed up (a // g) * b. */
+    if (PyObject_RichCompareBool(b, a, Py_LT)) {
```

Comment:
`PyObject_RichCompareBool()` can fail and set error.

What if arguments are negative?

Is there a practical difference between comparing values of integers and only their sizes?

Reply:
Oh thanks for the reminder, I would add some checks on the compared result.
On the other hand the comment https://github.com/python/cpython/issues/102221#issuecomment-1784049499 in the issue seems to agree with your point: it suffices to compare the sizes (bits) of them. Let me modify so.